backend/internal/cmds/import.go

```diff
@@ -0,0 +1,371 @@
+		return err
+	}
+
+	path := filepath.Join("migrations", string(common.EnvConfig.DbProvider))
+	source, err := iofs.New(resources.FS, path)
+	if err != nil {
+		return fmt.Errorf("failed to create embedded migration source: %w", err)
+	}
+
+	m, err := migrate.NewWithInstance("iofs", source, "pocket-id", driver)
+	if err != nil {
+		return fmt.Errorf("failed to create migration instance: %w", err)
+	}
+
+	if err := m.Migrate(targetVersion); err != nil && !errors.Is(err, migrate.ErrNoChange) {
+		if strings.HasPrefix(err.Error(), "no migration found") {
+			return fmt.Errorf("database version is newer than the latest supported version (%d) by the current Pocket ID version", targetVersion)
+		}
+		return fmt.Errorf("migration failed: %w", err)
+	}
+
+	return nil
+}
+
+// insertData populates the DB with the imported data
+func insertData(db *gorm.DB, dbData databaseJson) error {
+	// Disable foreign key checks for SQLite
+	if common.EnvConfig.DbProvider == common.DbProviderSqlite {
+		if err := toggleSqliteForeignKeyChecks(db, false); err != nil {
+			return fmt.Errorf("failed to disable foreign keys: %w", err)
+		}
+		defer func() {
+			if err := toggleSqliteForeignKeyChecks(db, true); err != nil {
+				fmt.Printf("Warning: failed to re-enable foreign keys: %v\n", err)
+			}
+		}()
+	}
+
+	return db.Transaction(func(tx *gorm.DB) error {
+		// Disable foreign key checks for Postgres
+		if common.EnvConfig.DbProvider == common.DbProviderPostgres {
+			if err := togglePostgresForeignKeyChecks(tx, false); err != nil {
+				return fmt.Errorf("failed to disable foreign keys: %w", err)
+			}
+			defer func() {
+				if err := togglePostgresForeignKeyChecks(tx, true); err != nil {
+					fmt.Printf("Warning: failed to re-enable foreign keys: %v\n", err)
+				}
+			}()
+		}
+
+		// Insert rows
+		for table, rows := range dbData.Tables {
+			if table == "schema_migrations" {
+				continue
+			}
+
+			for _, row := range rows {
+				normalizeRow(row)
+				if err := tx.Table(table).Create(row).Error; err != nil {
+					return fmt.Errorf("failed inserting into %s: %w", table, err)
+				}
+			}
+		}
+
+		return nil
+	})
+}
+
+// normalizeRow mutates a row so it round-trips correctly for SQLite
+func normalizeRow(row map[string]any) {
+	for k, v := range row {
+		switch val := v.(type) {
+		case map[string]any:
+			// Handle binary wrapper
+			if b64, ok := val["__binary__"].(string); ok {
+				if data, err := base64.StdEncoding.DecodeString(b64); err == nil {
+					row[k] = data
+					continue
+				}
+			}
+			// Otherwise encode as JSON string.
+			if b, err := json.Marshal(val); err == nil {
+				row[k] = string(b)
+			}
+
+		case []any:
+			if b, err := json.Marshal(val); err == nil {
+				row[k] = string(b)
+			}
+		}
+	}
+}
+
+// extractIntoBase writes a file entry from the ZIP under baseDir, removing the given prefix
+func extractIntoBase(f *zip.File, baseDir, stripPrefix string) error {
+	const maxFileSize = 50 << 20 // 50 MiB is a very generous limit
```

Comment:
What if someone has a lot of custom user profile images? I don't think we should put any max limit, since this isn't from a HTTP endpoint anyways